src/Webhook/Services/SubscriptionHandlerService.php

```diff
@@ -56,42 +64,24 @@ protected function handleCanceled(Webhook $webhook)
         return $result;
     }
 
-    protected function loadOrder(Webhook $webhook)
+    public function loadOrder(Webhook $webhook)
     {
-        $orderRepository = new OrderRepository();
+        $subscriptionRepository = new SubscriptionRepository();
+        $apiService = new ApiService();
 
-        /**
-         * @var Order $order
-         */
-        $order = $webhook->getEntity();
-        $order = $orderRepository->findByCode($order->getCode());
-        if ($order === null) {
-            $orderDecoratorClass = MPSetup::get(MPSetup::CONCRETE_PLATFORM_ORDER_DECORATOR_CLASS);
-
-            /**
-             * @var Order $webhookOrder
-             */
-            $webhookOrder = $webhook->getEntity();
-
-            /**
-             * @var PlatformOrderInterface $order
-             */
-            $order = new $orderDecoratorClass();
-            $order->loadByIncrementId($webhookOrder->getCode());
-
-            if ($order->getIncrementId() === null) {
-                throw new NotFoundException(
-                    "Order Not found!"
-                );
-            }
-
-            $orderFactory = new OrderFactory();
-            $order = $orderFactory->createFromPlatformData(
-                $order,
-                $webhookOrder->getMundipaggId()->getValue()
-            );
+        $subscriptionId = $webhook->getEntity()->getSubscriptionId()->getValue();
+        $subscriptionObject = $apiService->getSubscription(new SubscriptionId($subscriptionId));
+
+        if (is_null($subscriptionObject)) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function is_null() is discouraged; use strict comparison "=== null" instead.](https://app.codacy.com/manual/mundipagg/ecommerce-module-core/pullRequest?prid=4843277)